pkg/sql/err_count_test.go

```diff
@@ -81,3 +87,57 @@ func TestUnimplementedCounts(t *testing.T) {
 		t.Fatal("expected unimplemented telemetry, got nothing")
 	}
 }
+
+func TestTransactionRetryErrorCounts(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	telemetry.GetAndResetFeatureCounts(false)
+
+	// Transaction retry errors aren't given a pg error code until deep
+	// in pgwire (pgwire.convertToErrWithPGCode). Make sure we're
```

Comment:
I think the current appstats also have an attempt count of some sort but I don't know exactly what it means -- @knz might.

Reply:
I know even less about this (what's an appstat?) but if I'm understanding you correctly we want to count how many times we return these errors to the client even if the client doesn't retry (in a way that we can identify).